### DISCUSSION TITLE
Move Sharepoint Online Existing List items to another folder in the same list

### DISCUSSION THREAD

**Discussion by jomambas** [pnp/powershell#1991] [Q&A]
Hello!

I would like to use the pnp module to automate a project that I am working on. 
For new items, I can direct them to the folder where I want them to be created
add-pnplistitem -folder

However, the -folder option is not available when using set-pnplistitem.

Am I looking at the right direction? how do I move existing list items to folders in the same list? using pnp?

Thank you very much in advance!

**Comment by mharleydev** [comment]
@jomambas have you been able to find a solution to this?

**Reply by jomambas** [reply]
@mharleydev not a neat looking one. I used remove-pnplistitem for items that needed to be moved, then readded them with the -folder option.

**Reply by mharleydev** [reply]
I was able to make this work by using the server relative url of the list and the fileleafref of the item.

```powershell
  $ItemFileRef = $Item.FieldValues.FileLeafRef
  $SourceUrl = $List.RootFolder.ServerRelativeUrl + '/' +  $ItemFileRef
  $TargetUrl = $List.RootFolder.ServerRelativeUrl + '/' +  $NewFolderName + '/' +  $ItemFileRef
  Move-PnPFile -SourceUrl $SourceUrl -TargetUrl $TargetUrl -Force  
```
